handler/src/main/java/io/netty/handler/flush/FlushConsolidationHandler.java

```diff
@@ -70,17 +75,21 @@ public FlushConsolidationHandler(int explicitFlushAfterFlushes) {
     }
 
     @Override
+    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        this.ctx = ctx;
```

Comment:
If my understanding is correct, this is right as long as the handler is not shared.